PostPairsTrading.md

```diff
@@ -0,0 +1,319 @@
+superTab: `sym xgroup raze read_stock each syms
+```
+
+We then proceed to create a function called **fCoint** to call our imported function from PyKX, handle any null values by filling them with 0, using `0f^` and return the second value, which in this case is the p-value.
+
+```q
+fCoint: {@[;1]0f^coint[0f^x;0f^y]`}
+```
+
+We generate all combinations (`cross`) of indexes to see which pair is most cointegrated. Then, we index (`@`) each pair in our table. Additionally, we take (`#`) the last **trange** days of data for both indexes, and finally apply our **fCoint** function to each (`.'`) pair of data lists. **trange** symbolizes the number of working days in the last 4 years.
+
+
+```q
+trange:4*252
+matrix: fCoint .' neg[trange]#''@\:[;`close](@/:[superTab]')syms cross syms
+```
+
+Now, with our matrix in hand, we can plot it and **visually identify** which asset is more favorable. In order to do that, we can leverage PyKX once again to bring the `heatmap` module to q:
+
+```q
+pyhm:.pykx.import[`seaborn]`:heatmap
+pyhm[pvalues;`xticklabels pykw syms;`yticklabels pykw syms;`cmap pykw `RdYlGn_r]
+```
+
+And plot it:
+
+```q
+pyshow:.pykx.import[`matplotlib.pyplot]`:show
+pyshow[::]
+```
+
+Given our following assets:
+
+| SP500 | NASDAQ100 | BFX     | FCHI   | GDAXI   | HSI       | KS11    | MXX    | N100   | N225  | NYA | RUT | STOXX  |
+|:-----:|:---------:|:-------:|:------:|:-------:|:---------:|:-------:|:------:|:------:|:-----:|:---:|:---:|:------:|
+| USA   | USA       | Belgium | France | Germany | Hong Kong | S.Korea | Mexico | Europe | Japan | USA | USA | Europe |
+
+Our heatmap looks like this:
+
+![ADF heatmap](https://github.com/hablapps/pairstrading/blob/5-Post/resources/ADFgif.gif?raw=true)
+
+As we can observe, there are several cointegrated indices, but our attention will be drawn towards the **NASDAQ100 and SP500** synergy. Both of these indices belong to the American market and share numerous characteristics. They encompass American companies traded within the same scenario, which is what makes them a perfect fit for our case.
+In this heatmap, they exhibit a vibrant green color, indicative of a high degree of cointegration, or, in simpler terms, a very low probability of not being cointegrated. They demonstrate low p-values suggesting their strength as candidates.
+
+ > 💡 As we can see, this pair of indexes is not the best candidate according to our ADF tests. However, we chose it because the tick data for their prices is publicly available. We used TickStory to obtain the data.
+
+![Prices](https://github.com/hablapps/pairstrading/blob/5-Post/resources/Prices%20gif.gif?raw=true)
+
+
+The graphs illustrate the concept of cointegration between two indexes. The top two graphs show the prices of SP500 (left) and NASDAQ100 (right) over the same time period. We can observe that the price movements of these two indices follow similar patterns, suggesting some level of cointegration.

```

Comment:
So, what are `USA500` and `USATECH` exactly? What is the connection with these names?

Reply:
The official names are S&P 500 and NASDAQ-100. As far as I know, USA500 and USATECH are synonyms used by some brokers or financial platforms to refer to these indexes